src/app/components/SitePreferences/index.tsx

```diff
@@ -214,31 +224,97 @@ function SitePreferences({ launcherType, allowance, onEdit, onDelete }: Props) {
           </div>
 
           {hasPermissions && (
-            <div>
-              <h2 className="pt-4 text-lg text-gray-900 font-bold dark:text-white">
-                {t("edit_permissions")}
+            <div className="flex flex-col gap-4 pb-3 border-b border-gray-200 dark:border-neutral-700">
+              <h2 className="pt-5 text-md text-gray-800 dark:text-neutral-200">
+                {t("website_permissions")}
               </h2>
               <div>
-                {permissions.map((permission) => (
-                  <Fragment key={permission.id}>
-                    <Setting
-                      title={permission.method}
-                      inline={true}
-                      subtitle={tPermissions(
-                        permission.method
-                          .toLowerCase()
-                          .split("/")
-                          .slice(-2)
-                          .join(".") as unknown as TemplateStringsArray
-                      )}
-                      /* split the method at "/", take the last two items in
-                        the array and join them with "." to get the i18n string
-                        webln/lnd/getinfo -> lnd.getinfo
-                        nostr/nip04decrypt --> nostr.nip04decrypt */
-                    >
-                      <Toggle
-                        checked={permission.enabled}
-                        onChange={() => {
+                <p className="mb-3 text-xs font-semibold text-gray-800 dark:text-neutral-200">
+                  {t("edit_allowance.always_allow")}
+                </p>
+                <div>
+                  {permissions.map((permission) => (
+                    <Fragment key={permission.id}>
+                      <Badge
+                        key={permission.method}
+                        sitePermission={
+                          permission.method
+                            .toLowerCase()
```

Comment:
wondering if the handling of the method and this splitting and concatenation should somehow go into a function? 
we do this on several places  I think

Reply:
refactored this part of the logic. resolved